src/sentry/integrations/discord/utils/metrics.py

```diff
@@ -10,10 +15,19 @@
 
 
 def record_lifecycle_termination_level(lifecycle: EventLifecycle, error: ApiError) -> None:
+    try:
+        translate_discord_api_error(error)
+    except IntegrationConfigurationError as e:
+        lifecycle.record_halt(e)
+    except IntegrationError as e:
+        lifecycle.record_failure(e)
+
+
+def translate_discord_api_error(error: ApiError) -> None:
     if isinstance(error, ApiRateLimitedError):
         # TODO(ecosystem): We should batch this on a per-organization basis
-        lifecycle.record_halt(error)
+        raise IntegrationConfigurationError(error.text) from error
```

Comment:
Do you know if this is a per-discord guild rate limit? Or a global rate limit?

Depending on which one it is, we may still want to classify it as a hard failure, not a soft one.